### PR TITLE
update pico-examples/ide/vscode

### PR DESCRIPTION
**Pull request by mjcross**
1. replace deprecated uses of 'runToMain' with 'runToEntryPoint'
2. add example launcher for Picoprobe debugger
3. add simple readme

### CODE REVIEW

File: ide/vscode/README.md [hunk 1]
```diff
@@ -0,0 +1,11 @@
+# Sample configuration files for VSCode
+
+This example provides sample debug configurations that my be useful for developers using  the Microsoft(R) Visual Studio Code integrated development environment.
+
+A basic setup for Visual Studio Code is described in [Getting Started with the Raspberry Pi Pico](https://rptl.io/pico-get-started). More advanced configurations may need additional plug-ins such as *cortex-debug*.
```
Comment (lurch) [inline ide/vscode/README.md:5]: cortex-debug **is** mentioned in Chapter 7 of that PDF?

File: ide/vscode/launch-remote-openocd.json [hunk 2]
```diff
@@ -13,7 +13,7 @@
             // Connect to an already running OpenOCD instance
             "gdbTarget": "your-openocd:3333",
             "svdFile": "${env:PICO_SDK_PATH}/src/rp2040/hardware_regs/rp2040.svd",
-            "runToMain": true,
+            "runToEntryPoint": "Main",
```
Comment (lurch) [inline ide/vscode/launch-remote-openocd.json:16]: Should this be "main" with a lowercase 'm', to match the other .json files here? :man_shrugging:

File: ide/vscode/README.md (hunk 1 above)
Comment (mjcross) [reply ide/vscode/README.md]: Agreed - reference removed with [0153dafc](https://github.com/mjcross/pico-examples/commit/0153dafc773f6d3006fb85baa3f7c95a04f1484c)

File: ide/vscode/launch-remote-openocd.json (hunk 2 above)
Comment (mjcross) [reply ide/vscode/launch-remote-openocd.json]: Good spot! Resolved [0153dafc](https://github.com/mjcross/pico-examples/commit/0153dafc773f6d3006fb85baa3f7c95a04f1484c)

File: ide/vscode/README.md [hunk 3]
```diff
@@ -0,0 +1,11 @@
+# Sample configuration files for VSCode
+
+This example provides sample debug configurations that my be useful for developers using  the Microsoft(R) Visual Studio Code integrated development environment.
+
+A basic setup for Visual Studio Code is described in [Getting Started with the Raspberry Pi Pico](https://rptl.io/pico-get-started).
+
+To use the files copy them into a `.vscode` folder at the top level of your project and adapt them to your needs (see the [Visual Studio documentation](https://code.visualstudio.com/docs/cpp/launch-json-reference)).
```
Comment (lurch) [inline ide/vscode/README.md:7]: Sorry to nitpick, but is this "Visual Studio documentation" or "Visual Studio Code documentation" ?

File: ide/vscode/README.md [hunk 4]
```diff
@@ -0,0 +1,11 @@
...
+
+The files with names like `launch-*.json` illustrate different debugging configurations. You should select one of these, rename it to `launch.json` and make any desired modifications. Note that VS Code will ignore the others.
+
+The `settings.json` file illustrates how to configure the VS Code *CMake* plug-in to ensure that you debug using *cortex-debug* instead of trying to launch the execuable on the host.
```
Comment (lurch) [inline ide/vscode/README.md:11]: "execuable" -> "executable" ?

Comment (mjcross) [reply ide/vscode/README.md]: Thanks - fixed

File: ide/vscode/README.md (hunk 3 above)
Comment (mjcross) [reply ide/vscode/README.md]: No problem - not 'nitpicking' just effective peer review ;-) Fixed in [d5cb160](https://github.com/raspberrypi/pico-examples/pull/313/commits/d5cb160cf07741a4538f696f4161b4dae0d600a9)

**Comment by peterharperuk** [comment]
@lurch Are you taking care of this?

**Comment by kilograham** [comment]
@andygpz11 can you check this out if you have VSCode

**Comment by mjcross** [comment]
Thanks all.
Also is there a reason these examples set gdbPath to arm-multiarch? I didn’t want to mess with it but for individual developers I think it’s more likely to be arm-none-eabi so perhaps that should be the default.

**Comment by P33M** [comment]
Two comments:
- the `interface/picoprobe.cfg` openocd config command needs to be replaced with `interface/cmsis-dap.cfg` to correspond to the current instructions in the getting-started pdf
- arm-none-eabi-gdb is the tool installed by the ARM windows toolchain installer, but these instructions also apply to running on a pi - where gdb-multiarch is installed by default. So the default shouldn't really change unless this is a windows-specific guide, and that won't be the case as we're working on a separate method of getting the SDK up and running with VScode on windows.

**Comment by mjcross** [comment]
Thanks @P33M that’s useful to know.
1. The PR isn’t for a guide but for the existing examples in pico-examples/ide
2. The toolchain is also arm-none-eabi-gdb on MacOS which is what I’m using here. I’m not sure about Ubuntu/Debian.
I’ll have another look at the PR and propose something

**Review by andygpz11 — DISMISSED** [review]
I've pulled the changes here, copied the files from ide/vscode to .vscode and can confirm that I can program & debug using with these files via my picoprobe.
I did "hard code" the path to the executable to program and debug [since I don't use vscode to do the builds] but that's a one line change due to my machine set-up and not a line that this PR changes.
So, this looks good to me.
Thanks.

**Comment by mjcross** [comment]
Thanks @andygpz11  - I'm just improving the README and will push a commit to my local repo in the next half hour

**Comment by mjcross** [comment]
Proposed update to
- improve the README and comments, 
- fix the 'interface' setting issue identified by @P33M 
- separate out the default gdbPath setting where the toolchain is installed on a platform that's not a Raspberry Pi

**Review by mjcross — COMMENTED** [review]
Thanks @kilograham - definite improvements.
On reflection line 19 might be better without that 'just'; and I think 'further' should probably be after 'customise' (split infinitive)

File: ide/vscode/README.md [hunk 5]
```diff
@@ -0,0 +1,37 @@
+# Sample configuration files for developing with VSCode and an SWD Probe
+
+This example provides files that illustrate how to use Microsoft(R) Visual Studio Code (*VSCode*) with a Serial Wire Debug (*SWD*) probe such as another Pico running running [picoprobe](https://www.raspberrypi.com/documentation/microcontrollers/raspberry-pi-pico.html#debugging-using-another-raspberry-pi-pico).
```
Comment (lurch) [inline ide/vscode/README.md:3]: "running running" typo :wink:

File: ide/vscode/launch-raspberrypi-swd.json [hunk 6]
```diff
@@ -8,15 +8,15 @@
             "request": "launch",
             "type": "cortex-debug",
             "servertype": "openocd",
-            // This may need to be arm-none-eabi-gdb depending on your system
+            // This may need to be arm-none-eabi-gdb if VSCode is not running on a Raspberry Pi
             "gdbPath" : "gdb-multiarch",
             "device": "RP2040",
             "configFiles": [
                 "interface/raspberrypi-swd.cfg",
```
Comment (lurch) [inline ide/vscode/launch-raspberrypi-swd.json:15]: AIUI, this is for when the SWD pins on a Pico are connected directly to the GPIO pins on a Raspberry Pi (see page 19 of https://datasheets.raspberrypi.com/pico/getting-started-with-pico.pdf ), which doesn't match your "A target connected via another Pico" assumption in the README. I.e. you might want to clarify your README a bit further?

Comment (mjcross) [reply ide/vscode/launch-raspberrypi-swd.json:15]: Ah - didn't know that: thanks
Yes I'll tweak the README

File: ide/vscode/README.md (hunk 5 above)
Comment (mjcross) [reply ide/vscode/README.md]: Fixed in 27a065

File: ide/vscode/launch-raspberrypi-swd.json (hunk 6 above)
Comment (mjcross) [reply ide/vscode/launch-raspberrypi-swd.json:15]: @lurch comment really clarified my understanding of what's going on in the different configurations, so apologies for the further changes but I hope 27a065 is an improvement.

**Review by kilograham — APPROVED** [review]

**Pull request merged**